### ISSUE TITLE
HttpClient SendAsync sending a GET instead of DELETE on Mac OS

### ISSUE THREAD

**Issue by Gabez0r** [dotnet/corefx#4134]
When I make a DELETE request via SendAsync, a GET is sent instead of a DELETE. This happens only on Mac OS. The code produces a correct DELETE request on Windows.

Code:
```C#
using System;
using System.Net.Http;

namespace Deltest
{
	public class Program
	{
		public static void Main(string[] args)
		{
			HttpClient client = new HttpClient(new HttpClientHandler()
            {
				// redirect to Charles Proxy on Mac OS and Fiddler on Windows
                Proxy = new WebProxy(new Uri("http://127.0.0.1:8888")),
                UseProxy = true
            });
			HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Delete, "http://localhost:8080/document"); // on Windows the host's IP is used instead
			HttpResponseMessage response = client.SendAsync(request).Result;
			Console.WriteLine(response);
		}	
		
	}
}
```

On Mac OS El Capitan, the following HTTP Request is send:
```
GET /document HTTP/1.1
Host: localhost:8080
Accept: */*
Accept-Encoding: gzip,deflate
Cookie: 
```

Whereas on Windows 10, it is as follows:
```
DELETE http://192.168.1.8:8080/document HTTP/1.1
Accept-Encoding: gzip, deflate
Connection: Keep-Alive
Host: 192.168.1.8:8080
```

Mac OS is the host machine. The code was run from both the host machine and a Windows client VM, connecting to the same server running on the host.

Is this a bug or am I missing something?

Thank you for your time.

**Comment by stephentoub** [comment]
Thanks, @Gabez0r.  This looks like a bug in the Unix implementation.

**Comment by kapilash** [comment]
Thanks, @Gabez0r 
This is fixed via #4166

**Comment by kapilash** [comment]
This fix has not been merged into release/1.0.0-rc1. So reopenining it.

**Comment by kapilash** [comment]
Fixed via #4199
